subsys/bluetooth/host/hci_core.c

```diff
@@ -6692,3 +6713,20 @@ int bt_le_oob_get_sc_data(struct bt_conn *conn,
 	return bt_smp_le_oob_get_sc_data(conn, oobd_local, oobd_remote);
 }
 #endif
+
+#if defined(CONFIG_BT_PRIVACY)
+int bt_set_rpa_timeout(u16_t timeout)
+{
+	int err;
+
+	/* Restart timer even with new value */
```

Comment:
This is a restart, but no action taken? Then it would be remaning_value + new_value until RPA timeout happens again. I think it should be refreshed immediately here, and then again in new_value in the future.

If we refresh RPA here we need to handle conn create using the current RPA, as the random address cannot be updated while the controller initiatior is active.

Reply:
> This is a restart, but no action taken? Then it would be remaning_value + new_value until RPA timeout happens again. I think it should be refreshed immediately here, and then again in new_value in the future.
> 
> If we refresh RPA here we need to handle conn create using the current RPA, as the random address cannot be updated while the controller initiatior is active.

Do you mean "elapsed + new"? Then yes, that would potentially extend the RPA refresh. It would be a good idea to refresh the RPA immediately or by using a short "inital work delay" to have the regular timeout do it and restart with new value.
I'm not sure I can see how to prevent RPA refresh while controller initiatior is active in all cases?